usm_rest_api/v1/views/utils.py

```diff
@@ -0,0 +1,267 @@
+            clusterSerilaizer.save()
+        else:
+                log.error(
+                    "Cluster Creation failed. Invalid clusterSerilaizer")
+                log.error(
+                    "clusterSerilaizer Err: %s" % clusterSerilaizer.errors)
+                raise Exception(
+                    "Cluster Creation failed. Invalid clusterSerilaizer",
+                    clusterSerilaizer.errors)
+    except Exception, e:
+        log.exception(e)
+        raise e
+
+
+def setup_transport_and_update_db(cluster_data, nodelist):
+    log.debug("Inside setup_transport_and_update_db function")
+    failedNodes = []
+
+    # Create the Nodes in the cluster
+    # Setup the host communication channel
+    # current_task.update_state(state='ESTABLISH_HOST_COMMUNICATION')
+    minionIds = {}
+    for node in nodelist:
+        try:
+            ssh_fingerprint = usm_wrapper_utils.get_host_ssh_key(
+                node['management_ip'])
+            minionIds[node['management_ip']] = salt_wrapper.setup_minion(
+                node['management_ip'], ssh_fingerprint[0],
+                node['ssh_username'], node['ssh_password'])
+        except Exception, e:
+            log.exception(e)
+            failedNodes.append(node)
+    log.debug(minionIds)
+    # Sleep for sometime to make sure all the restarted minions are back
+    # online
+    # TODO - Sleep will be removed from here. Need to look at the events from
+    # salt to make sure that the channel is ready for sending the commands
+    time.sleep(ACCEPT_MINION_TIMEOUT)
+    log.debug("Accepting the minions keys")
+    # current_task.update_state(state='ADD_MINION_KEYS_AND_DB_UPDATE')
+
+    # Accept the keys of the successful minions and add to the DB
+    for node in nodelist:
+        try:
+            if node not in failedNodes:
+                salt_wrapper.accept_minion(minionIds[node['management_ip']])
+        except Exception, e:
+            log.exception(e)
+            failedNodes.append(node)
+    log.debug(minionIds)
+    #
+    # Wait for some time so that the communication chanel is ready
+    # TODO - Sleep will be removed from here. Need to look at the events from
+    # salt to make sure that the channel is ready for sending the commands
+    time.sleep(CONFIG_PUSH_TIMEOUT)
+
+    # Persist the hosts into DB
+    for node in nodelist:
+        #
+        # Get the host uuid from  host and update in DB
+        #
+        try:
+            if node not in failedNodes:
+                node['node_id'] = salt_wrapper.get_machine_id(
+                    minionIds[node['management_ip']])
+                node['cluster'] = str(cluster_data['cluster_id'])
+                hostSerilaizer = HostSerializer(data=node)
+
+                if hostSerilaizer.is_valid():
+                    # Delete all the fields those are not
+                    # reqired to be persisted
+                    del hostSerilaizer.validated_data['ssh_password']
```

Comment:
Actually those are writeable fields and serializer doenot expecct them to present and validate will fail. Other way to overrride serializer create method to handle this. Since I dont have anything else to do there, I prefer this.